### PR TITLE
Refactor OpenShift tests to work also with remote OCP4

### PR DESCRIPTION
**Pull request by hhorak**
This is a rather bigger re-factoring that I failed to do in smaller chunks. Hopefully tests passing in all the container images would be good enough proof it is ok to merge. I'll add links to particulars PRs in all container images repos:

- https://github.com/sclorg/s2i-python-container/pull/381
- https://github.com/sclorg/s2i-ruby-container/pull/270
- https://github.com/sclorg/s2i-nodejs-container/pull/240
- https://github.com/sclorg/s2i-perl-container/pull/177
- https://github.com/sclorg/s2i-php-container/pull/289
- https://github.com/sclorg/mariadb-container/pull/126
- https://github.com/sclorg/mysql-container/pull/288
- https://github.com/sclorg/postgresql-container/pull/378
- https://github.com/sclorg/mongodb-container/pull/319
- https://github.com/sclorg/redis-container/pull/60
- https://github.com/sclorg/httpd-container/pull/95
- https://github.com/sclorg/nginx-container/pull/112
- https://github.com/sclorg/varnish-container/pull/63

### CODE REVIEW

**Comment by hhorak** [comment]
@phracek @pkubatrh All tests for other container images passed, so I believe it is ready to merge now. I hope that passing tests are good enough as a review, but if you want to check it quickly, it would be even better.

**Comment by hhorak** [comment]
> I believe it is ready to merge now.

After merging #152, I rebased this PR and thus also common sub-module in other containers, at least we can see whether the tests pass again. Still ok to merge this one I believe.

**Comment by praiskup** [comment]
[test]

File: test-lib-openshift.sh [hunk 1]
```diff
@@ -788,15 +794,24 @@ function ct_os_test_template_app_func() {
...
 
   echo "  Checking APP using $check_command_exp ..."
   local result=0
   eval "$check_command_exp" || result=1
 
+  echo "  Information about the image we work with:"
+  oc get deploymentconfig.apps.openshift.io/"${service_name}" -o yaml | grep lastTriggeredImage
+  # for s2i builds, the resulting image is actually in the current namespace
+  oc get isimage -n "${namespace}" "${image_id##*/}" -o yaml || oc get isimage "${image_id##*/}" -o yaml
+
```
Comment (pkubatrh) [inline test-lib-openshift.sh:814]: This seems like it would be nice to have as a separate function

Comment (hhorak) [reply test-lib-openshift.sh:814]: good idea, will change it

Comment (hhorak) [reply test-lib-openshift.sh]: Done.

**Pull request merged**